src/Microsoft.Sbom.Api/Executors/ComponentDetectionBaseWalker.cs

```diff
@@ -32,6 +35,9 @@ public abstract class ComponentDetectionBaseWalker
     private readonly ISbomConfigProvider sbomConfigs;
     private readonly IFileSystemUtils fileSystemUtils;
 
+    public readonly Dictionary<string, string> LicenseDictionary = new Dictionary<string, string>();
```

Comment:
Is this class created as a singleton or on an as-needed basis? I ask because if a Dictionary is accessed in a multi-threaded way it can cause all kinds of exceptions.